xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp

```diff
@@ -16,6 +16,71 @@
 #include "cores/VideoPlayer/Interface/TimingConstants.h"
 #include "utils/log.h"
 
+namespace
+{
+AVPixelFormat ConvertPixFormat(VIDEOCODEC_FORMAT videoFormat)
+{
+  switch (videoFormat)
+  {
+    case VIDEOCODEC_FORMAT_YV12:
+      return AV_PIX_FMT_YUV420P;
+    case VIDEOCODEC_FORMAT_I420:
+      return AV_PIX_FMT_YUV420P;
+    case VIDEOCODEC_FORMAT_YUV420P9:
+      return AV_PIX_FMT_YUV420P9;
+    case VIDEOCODEC_FORMAT_YUV420P10:
+      return AV_PIX_FMT_YUV420P10;
+    case VIDEOCODEC_FORMAT_YUV420P12:
+      return AV_PIX_FMT_YUV420P12;
+    case VIDEOCODEC_FORMAT_YUV422P9:
+      return AV_PIX_FMT_YUV422P9;
+    case VIDEOCODEC_FORMAT_YUV422P10:
+      return AV_PIX_FMT_YUV422P10;
+    case VIDEOCODEC_FORMAT_YUV422P12:
+      return AV_PIX_FMT_YUV422P12;
+    case VIDEOCODEC_FORMAT_YUV444P9:
+      return AV_PIX_FMT_YUV444P9;
+    case VIDEOCODEC_FORMAT_YUV444P10:
+      return AV_PIX_FMT_YUV444P10;
+    case VIDEOCODEC_FORMAT_YUV444P12:
+      return AV_PIX_FMT_YUV444P12;
+    default:
+      CLog::LogF(LOGWARNING,
+                 "CInputStreamAddon: Video pixel format '{}' not valid, fallback to YUV420P.",
+                 videoFormat);
+      return AV_PIX_FMT_YUV420P;
+  }
+}
+
+unsigned int GetColorBitsFromVideoFormat(VIDEOCODEC_FORMAT videoFormat)
+{
+  switch (videoFormat)
+  {
+    case VIDEOCODEC_FORMAT_YV12:
+    case VIDEOCODEC_FORMAT_I420:
+      return 8;
+    case VIDEOCODEC_FORMAT_YUV420P9:
+    case VIDEOCODEC_FORMAT_YUV422P9:
+    case VIDEOCODEC_FORMAT_YUV444P9:
+      return 9;
+    case VIDEOCODEC_FORMAT_YUV420P10:
+    case VIDEOCODEC_FORMAT_YUV422P10:
+    case VIDEOCODEC_FORMAT_YUV444P10:
+      return 10;
+    case VIDEOCODEC_FORMAT_YUV420P12:
+    case VIDEOCODEC_FORMAT_YUV422P12:
+    case VIDEOCODEC_FORMAT_YUV444P12:
+      return 12;
+    default:
+      CLog::LogF(LOGWARNING,
+                 "CInputStreamAddon: Video pixel format '{}' not valid, fallback to 8 bit.",
```

Comment:
"CInputStreamAddon" - this looks wrong. This is a free function, thus not bound to a class. Maybe use `CAddonVideoCodec`?